### ISSUE TITLE
Tons of exceptions while working with xml files

### ISSUE THREAD

**Issue by deathaxe** [rosshadden/sublime-xpath#28]
Sublime Text 3 3131 x64 on Windows 10 x64

lxml 3.4.4

Please catch parsing exceptions as they break event handling of other plugins.

```
Traceback (most recent call last):
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 30, in settingsChanged
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 256, in updateStatusToCurrentXPathIfSGML
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 105, in ensureTreeCacheIsCurrent
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 69, in buildTreesForView
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 81, in buildTreeForViewRegion
  File "lxml_parser in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 196, in lxml_etree_parse_xml_string_with_location
  File "lxml_parser in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 83, in feed
  File "lxml_parser in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 89, in _feed
  File "parser.pxi", line 1201, in lxml.etree._FeedParser.feed (src\lxml\lxml.etree.c:102246)
  File "parser.pxi", line 1325, in lxml.etree._FeedParser.feed (src\lxml\lxml.etree.c:102127)
  File "parsertarget.pxi", line 136, in lxml.etree._TargetParserContext._handleParseResult (src\lxml\lxml.etree.c:116979)
  File "parsertarget.pxi", line 130, in lxml.etree._TargetParserContext._handleParseResult (src\lxml\lxml.etree.c:116869)
  File "lxml.etree.pyx", line 316, in lxml.etree._ExceptionContext._raise_if_stored (src\lxml\lxml.etree.c:10323)
  File "saxparser.pxi", line 544, in lxml.etree._handleSaxPI (src\lxml\lxml.etree.c:112386)
  File "parsertarget.pxi", line 89, in lxml.etree._PythonSaxParserTarget._handleSaxPi (src\lxml\lxml.etree.c:116391)
TypeError: <lambda>() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "C:\Apps\Sublime\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 410, in on_activated_async
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 262, in updateStatusToCurrentXPathIfSGML
KeyError: 16
Traceback (most recent call last):
  File "C:\Apps\Sublime\sublime_plugin.py", line 513, in on_selection_modified_async
    callback.on_selection_modified_async(v)
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 407, in on_selection_modified_async
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 262, in updateStatusToCurrentXPathIfSGML
KeyError: 16
Traceback (most recent call last):
  File "C:\Apps\Sublime\sublime_plugin.py", line 513, in on_selection_modified_async
    callback.on_selection_modified_async(v)
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 407, in on_selection_modified_async
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 262, in updateStatusToCurrentXPathIfSGML
KeyError: 16
Traceback (most recent call last):
  File "C:\Apps\Sublime\sublime_plugin.py", line 513, in on_selection_modified_async
    callback.on_selection_modified_async(v)
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 407, in on_selection_modified_async
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 262, in updateStatusToCurrentXPathIfSGML
KeyError: 16
Traceback (most recent call last):
  File "C:\Apps\Sublime\sublime_plugin.py", line 513, in on_selection_modified_async
    callback.on_selection_modified_async(v)
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 407, in on_selection_modified_async
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 262, in updateStatusToCurrentXPathIfSGML
KeyError: 16
Traceback (most recent call last):
  File "C:\Apps\Sublime\sublime_plugin.py", line 513, in on_selection_modified_async
    callback.on_selection_modified_async(v)
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 407, in on_selection_modified_async
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 262, in updateStatusToCurrentXPathIfSGML
KeyError: 16
Traceback (most recent call last):
  File "C:\Apps\Sublime\sublime_plugin.py", line 513, in on_selection_modified_async
    callback.on_selection_modified_async(v)
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 407, in on_selection_modified_async
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 262, in updateStatusToCurrentXPathIfSGML
KeyError: 16
Traceback (most recent call last):
  File "C:\Apps\Sublime\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 410, in on_activated_async
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 262, in updateStatusToCurrentXPathIfSGML
KeyError: 16
Traceback (most recent call last):
  File "C:\Apps\Sublime\sublime_plugin.py", line 530, in on_activated_async
    callback.on_activated_async(v)
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 410, in on_activated_async
  File "xpath in C:\Apps\Sublime\Data\Installed Packages\xpath.sublime-package", line 262, in updateStatusToCurrentXPathIfSGML
KeyError: 16
```

The following xml file caused the errors.

```XML
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://www.oxygenxml.com/docbook/xml/5.0/rng/dbmathmlsvg.rng" schematypens="http://relaxng.org/ns/structure/1.0"?>
<article xmlns="http://docbook.org/ns/docbook" version="5.0"
    xmlns:xlink="http://www.w3.org/1999/xlink">
    <title>MathML and SVG</title>
    <sect1>
        <title>MathML</title>
        <sect2>
            <title>Embedded MathML </title>
            <para>The equation as a block. A line of text.</para>
            <equation>
                <mml:math xmlns:mml="http://www.w3.org/1998/Math/MathML" display="block">
                    <mml:mrow>
                        <mml:msub>
                            <mml:mi>f</mml:mi>
                            <mml:mi>n</mml:mi>
                        </mml:msub>
                        <mml:mo>=</mml:mo>
                        <mml:mfrac>
                            <mml:mn>1</mml:mn>
                            <mml:msqrt>
                                <mml:mn>5</mml:mn>
                            </mml:msqrt>
                        </mml:mfrac>
                        <mml:mrow>
                            <mml:mo>[</mml:mo>
                            <mml:mrow>
                                <mml:msup>
                                    <mml:mrow>
                                        <mml:mo>(</mml:mo>
                                        <mml:mfrac>
                                            <mml:mrow>
                                                <mml:mn>1</mml:mn>
                                                <mml:mo>+</mml:mo>
                                                <mml:msqrt>
                                                  <mml:mn>5</mml:mn>
                                                </mml:msqrt>
                                            </mml:mrow>
                                            <mml:mn>2</mml:mn>
                                        </mml:mfrac>
                                        <mml:mo>)</mml:mo>
                                    </mml:mrow>
                                    <mml:mi>n</mml:mi>
                                </mml:msup>
                                <mml:mo>-</mml:mo>
                                <mml:msup>
                                    <mml:mrow>
                                        <mml:mo>(</mml:mo>
                                        <mml:mfrac>
                                            <mml:mrow>
                                                <mml:mn>1</mml:mn>
                                                <mml:mo>-</mml:mo>
                                                <mml:msqrt>
                                                  <mml:mn>5</mml:mn>
                                                </mml:msqrt>
                                            </mml:mrow>
                                            <mml:mn>2</mml:mn>
                                        </mml:mfrac>
                                        <mml:mo>)</mml:mo>
                                    </mml:mrow>
                                    <mml:mi>n</mml:mi>
                                </mml:msup>
                            </mml:mrow>
                            <mml:mo>]</mml:mo>
                        </mml:mrow>
                    </mml:mrow>
                </mml:math>
            </equation>
            <para>Another line of text.</para>
            <para>Inline equation. The equation :<inlineequation><mml:math
                        xmlns:mml="http://www.w3.org/1998/Math/MathML">
                        <mml:mstyle>
                            <mml:mrow>
                                <mml:mfrac>
                                    <mml:mrow>
                                        <mml:mi>sin</mml:mi>
                                        <mml:mo rspace="verythinmathspace">&#x2061;</mml:mo>
                                        <mml:mi>θ</mml:mi>
                                    </mml:mrow>
                                    <mml:mi>π</mml:mi>
                                </mml:mfrac>
                            </mml:mrow>
                        </mml:mstyle>
                    </mml:math>
                </inlineequation> followed by some more text.</para>
        </sect2>
        <sect2>
            <title>External MathML Equation</title>
            <para>Referred using the <tag>imagedata</tag> element.</para>
            <para>
                <mediaobject>
                    <imageobject>
                        <imagedata fileref="../../mathml/sample.mathml"/>
                    </imageobject>
                </mediaobject>
            </para>
        </sect2>
    </sect1>
    <sect1>
        <title>SVG</title>
        <sect2>
            <title>Embedded SVG Images</title>
            <para>The SVG content is wrapped into an <tag>imageobject</tag> element.</para>
            <figure>
                <title>An SVG figure. </title>
                <mediaobject>
                    <imageobject>
                        <imagedata>
                            <svg:svg width="5cm" height="4cm" viewBox="0 0 500 400" version="1.1"
                                xmlns:svg="http://www.w3.org/2000/svg">
                                <svg:title>Example cubic01- cubic Bezier commands in path
                                    data</svg:title>
                                <svg:desc>Picture showing a simple example of path data using both a
                                    "C" and an "S" command, along with annotations showing the
                                    control points and end points</svg:desc>
                                <svg:style type="text/css"> .Border { fill:none; stroke:blue;
                                    stroke-width:1 } .Connect { fill:none; stroke:#888888;
                                    stroke-width:2 } .SamplePath { fill:none; stroke:red;
                                    stroke-width:5 } .EndPoint { fill:none; stroke:#888888;
                                    stroke-width:2 } .CtlPoint { fill:#888888; stroke:none }
                                    .AutoCtlPoint { fill:none; stroke:blue; stroke-width:4 } .Label
                                    { font-size:22; font-family:Verdana } </svg:style>
                                <svg:rect class="Border" x="1" y="1" width="498" height="398"/>
                                <svg:polyline class="Connect" points="100,200 100,100"/>
                                <svg:polyline class="Connect" points="250,100 250,200"/>
                                <svg:polyline class="Connect" points="250,200 250,300"/>
                                <svg:polyline class="Connect" points="400,300 400,200"/>
                                <svg:path class="SamplePath"
                                    d="M100,200 C100,100 250,100 250,200 S400,300 400,200"/>
                                <svg:circle class="EndPoint" cx="100" cy="200" r="10"/>
                                <svg:circle class="EndPoint" cx="250" cy="200" r="10"/>
                                <svg:circle class="EndPoint" cx="400" cy="200" r="10"/>
                                <svg:circle class="CtlPoint" cx="100" cy="100" r="10"/>
                                <svg:circle class="CtlPoint" cx="250" cy="100" r="10"/>
                                <svg:circle class="CtlPoint" cx="400" cy="300" r="10"/>
                                <svg:circle class="AutoCtlPoint" cx="250" cy="300" r="9"/>
                                <svg:text class="Label" x="25" y="70">M100,200 C100,100 250,100
                                    250,200</svg:text>
                                <svg:text class="Label" x="325" y="350" style="text-anchor:middle"
                                    >S400,300 400,200</svg:text>
                            </svg:svg>
                        </imagedata>
                    </imageobject>
                </mediaobject>
            </figure>
            <para>This is an inline <inlinemediaobject><imageobject><imagedata><svg:svg
                                xmlns:svg="http://www.w3.org/2000/svg" width="3cm" height="3cm"
                                viewBox="0 0 400 400">
                                <svg:g style="fill-opacity:0.7; stroke:black; stroke-width:0.1cm;">
                                    <svg:circle cx="6cm" cy="2cm" r="100" style="fill:red;"
                                        transform="translate(0,50)"/>
                                    <svg:circle cx="6cm" cy="2cm" r="100" style="fill:blue;"
                                        transform="translate(70,150)"/>
                                    <svg:circle cx="6cm" cy="2cm" r="100" style="fill:green;"
                                        transform="translate(-70,150)"/>
                                </svg:g>
                            </svg:svg></imagedata></imageobject></inlinemediaobject> image.</para>
        </sect2>
        <sect2>
            <title>External SVG Images</title>
            <para>These images are referred using the <tag class="attribute">fileref</tag> attribute
                of the <tag>imageobject</tag> element.</para>
            <para>An image scaled down to 50%: <mediaobject>
                    <imageobject>
                        <imagedata fileref="../../svg/shapes.svg" scale="50" format="SVG"/>
                    </imageobject>
                </mediaobject> A complex transparent image that is not scaled: <mediaobject>
                    <imageobject>
                        <imagedata fileref="../../svg/blured-shapes.svg" format="SVG"/>
                    </imageobject>
                </mediaobject></para>
        </sect2>
    </sect1>
</article>
```

**Comment by keith-hall** [comment]
Thanks for the report - it parses fine (no exceptions) for me on Windows 7x64 with ST build 3131, but I'll take a deeper look when I get time and try to work out how to fix it for you.

**Comment by keith-hall** [comment]
Okay, I take that back - I can replicate it - it turns out that I've been using the version from my fork here, where it seems to have been fixed: https://github.com/keith-hall/sublime-xpath/tree/dbcd198ccc99a8fa54c002c1c4a586bdc227b696

I will try to take a look at making a PR to this repo :)

**Comment by keith-hall** [comment]
Also it seems the XML syntax marks the `xml-model` processing instruction as invalid, the same like I reported & fixed for `xml-stylesheet` - https://github.com/sublimehq/Packages/issues/882. It'd be great to get a full list of these, to make sure we fix them all :)

**Comment by deathaxe** [comment]
Don't have such a list unfortunately. I don't handle XML stuff everyday. Last time I had to write some xslt I used _Oxygen XML Editor_, but I actually hate software built on Java on Windows.

So I am looking for some packages which make working with DOCBOOK or DITA in ST bearable. xpath is one of them. I actually was trying to get _Exalt_ to work, when this issue raised here. But I disabled it after all as it seems unable to parse some of the DTDs well.
